libgalois/src/analytics/louvain_clustering/louvain_clustering.cpp

```diff
@@ -0,0 +1,615 @@
+          if (!curr_mod_result) {
+            return curr_mod_result.error();
+          }
+          curr_mod = curr_mod_result.value();
+          break;
+        }
+        default:
+          return katana::ErrorCode::InvalidArgument;
+        }
+      }
+
+      [[maybe_unused]] uint64_t num_unique_clusters =
+          Base::RenumberClustersContiguously(&graph_curr);
+
+      if (iter < plan.max_iterations() &&
+          (curr_mod - prev_mod) > plan.threshold()) {
+        if (!plan.is_enable_vf() && phase == 1) {
+          KATANA_LOG_DEBUG_ASSERT(num_nodes_orig == graph_curr.num_nodes());
+          katana::do_all(katana::iterate(graph_curr), [&](GNode n) {
+            clusters_orig[n] =
+                graph_curr.template GetData<LCCurrentCommunityId>(n);
+          });
+        } else {
+          katana::do_all(
+              katana::iterate((uint64_t)0, num_nodes_orig), [&](GNode n) {
+                if (clusters_orig[n] != Base::UNASSIGNED) {
+                  KATANA_LOG_DEBUG_ASSERT(
+                      clusters_orig[n] < graph_curr.num_nodes());
+                  clusters_orig[n] =
+                      graph_curr.template GetData<LCCurrentCommunityId>(
+                          clusters_orig[n]);
+                }
+              });
+        }
+
+        auto coarsened_graph_result =
+            Base::template GraphCoarsening<NodeData, EdgeData, EdgeWeightType>(
+                graph_curr, pfg_curr.get(), num_unique_clusters);
+        if (!coarsened_graph_result) {
+          return coarsened_graph_result.error();
+        }
+
+        pfg_curr = std::move(coarsened_graph_result.value());
+
+        prev_mod = curr_mod;
+        // Base::PrintGraphCharateristics(*(pfg_curr.get()));
+      } else {
+        break;
+      }
+    }
+    return katana::ResultSuccess();
+  }
+};
+
+template <typename EdgeWeightType>
+static katana::Result<void>
+LouvainClusteringWithWrap(
+    katana::PropertyGraph* pfg, const std::string& edge_weight_property_name,
+    const std::string& output_property_name, LouvainClusteringPlan plan) {
+  static_assert(
+      std::is_integral_v<EdgeWeightType> ||
+      std::is_floating_point_v<EdgeWeightType>);
+
+  using Impl = LouvainClusteringImplementation<EdgeWeightType>;
+  if (auto result = ConstructNodeProperties<typename Impl::NodeData>(pfg);
+      !result) {
+    return result.error();
+  }
+
+  /*
+   * To keep track of communities for nodes in the original graph.
+   * Community will be set to -1 for isolated nodes
+   */
+  LargeArray clusters_orig;
+  clusters_orig.allocateBlocked(pfg->num_nodes());
+
+  LouvainClusteringImplementation<EdgeWeightType> impl{};
+  if (auto r = impl.LouvainClustering(
+          pfg, edge_weight_property_name, clusters_orig, plan);
+      !r) {
+    return r.error();
+  }
+
+  // Remove all the existing node/edge properties
+  if (auto r = pfg->RemoveAllNodeProperties(); !r) {
+    return r.error();
+  }
```

Comment:
I don't think dropping all properties of a users graph is a good idea. That is not something anyone would expect a graph algorithm to do.

If you need to clean up from the algorithms mess maybe a vector of [TemporaryPropertyGuard](https://github.com/KatanaGraph/katana/blob/bc9a6c665bc8aeb1920c3ebb8567a6563c5162d3/libgalois/include/katana/analytics/Utils.h#L74) would be a simple way to handle it. When the vector destructs all the properties will delete. It could be scoped to this function.